src/core/renderer/utilities/AFManager.js

```diff
@@ -0,0 +1,43 @@
+class AFManager {
+
+	constructor() {
+
+		// XR session
+		this.xrsession = null;
+
+	}
+
+	// Set active XR session
+	setXRSession( xrsession ) {
+
+		this.xrsession = xrsession;
+
+	}
+
+	removeXRSession() {
+
+		this.xrsession = null;
+
+	}
```

Comment:
When changing the system being used for callbacks we will have to cancel all previous callbacks and reregister them with the new function. Eg if the "LRUCache" schedules an unload event it will be waiting until that event fires before scheduling another one. So if the last event is scheduled using the "xrSession" callback and then the session closes, the scheduled callback will never be fired.

So we'll need to keep track of all scheduled callbacks and rAF handles to swap over the scheduled events.